tests/metrics/metric_manager/test_metric_manager_ddp.py

```diff
@@ -0,0 +1,147 @@
+import unittest
+import os
+from src.constructor import METRICS
+from src.metrics.metric_manager import MetricParams, MetricManager, Phase
+
+from pytorch_lightning import LightningModule, Trainer
+import torch
+from torch import nn
+from torch.nn import functional as F
+from torch.utils.data import DataLoader, Dataset, random_split
+from torchmetrics import Metric, Accuracy
+
+from torchvision import transforms
+from torchvision.datasets import MNIST
+
+from typing import *
+
+INPUT_DATA_SHAPE = 16
+BATCH_SIZE = 4
+EPOCH = 5
+
+labels = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9,
+        9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9
+    ]
+
+predicts = [0, 0, 1, 3, 3, 4, 5, 6, 7, 0, 0, 1, 1, 2, 3, 3, 3, 3, 3, 3, 4, 4, 4, 5, 5, 6, 6, 7, 7, 
+     8, 8, 9, 9, 7, 7, 8, 8, 8, 8, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 9
+]
+
+uniq_label_count = 10
+accuracy_answer = 0.18
+
+
+class FakeData(Dataset):
+    def __init__(self, labels=labels, predicts=predicts):
+        self.labels = labels
+        self.predicts = predicts
+
+    def __len__(self):
+        return len(self.labels)
+
+    def __getitem__(self, item):
+        return torch.rand(INPUT_DATA_SHAPE), torch.tensor(self.predicts[item]), torch.tensor(self.labels[item])
+
+
+@METRICS.register_class
+class MetricMemoryBlock(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state("memory_list", default=[], dist_reduce_fx=None)
+
+    def update(self, state: torch.Tensor):
+        self.memory_list.append(state)
+
+    def compute(self):
+        return torch.tensor((torch.cat(self.memory_list)).shape[0])
+
+
+class Model(LightningModule):
+    def __init__(self, metric_params: List[MetricParams]):
+        super().__init__()
+        self.l1 = torch.nn.Linear(INPUT_DATA_SHAPE, uniq_label_count)
+        self.metric_manager = MetricManager(metric_params)
+
+    def forward(self, x):
+        return torch.relu(self.l1(x.view(x.size(0), -1)))
```

Comment:
Why not to mock the model and training steps?

Reply:
How to mock PL Module?